src/library_pthread.js

```diff
@@ -7,6 +7,7 @@
 var LibraryPThread = {
   $PThread__postset: 'if (!ENVIRONMENT_IS_PTHREAD) PThread.initMainThreadBlock();',
   $PThread__deps: ['_emscripten_thread_init',
+                   'emscripten_is_main_browser_thread',
```

Comment:
this function is in compiled code, and the emcc.py change ensures it is added and exported? Then this line should not be needed.

Reply:
I was looking at few other example of this yesterday.  For example, should we duplicate the dependencies in deps_info in the library JS files where possible. 

In general listing native dependencies in JS library files it now useful because they cannot be effective at actually bringing in those dependencies (since they act after the wasm-ld phase).

However, I think they can be useful.   A missing dependency here in the library code will cause a link time failure with nice diagnostic message.   Missing dependencies in `emcc.py` doesn't seem to do that.   I think the reason is that functions added by emcc.py to `EXPORTED_FUNCTIONS` are allowed to be missing at link time (unlike user-exported functions).  This is the same with deps_info.py I believe.

So maybe its OK/advisable to list JS->native dependencies listed in library JS files?  But maybe that should be discussed separately to this issue?

Reply:
We can discuss it separately, but yes, I agree with what you say. If we want to move to adding those deps, they would lead to better errors.

Reply:
I found this very useful during development to add this here because adding it to EXPORTED_FUNCTIONS during emcc.py wasn't enough to cause and error when it was missing.   So I'd like to keep this I think.